sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -571,6 +639,81 @@ def convert_feature(fx):
             )
         features = tmp_features
 
+        if categorical_features is None:
+            is_categorical = [
+                (False,) if len(fxs) == 1 else (False, False) for fxs in features
+            ]
+        else:
+            # we need to create a boolean indicator of which feature are
+            # categorical from the categorical_features list.
+            categorical_features = np.array(categorical_features, copy=False)
+            if categorical_features.dtype.kind == "b":
+                # categorical features provided as a list of boolean
+                if categorical_features.size != n_features:
+                    raise ValueError(
+                        "When `categorical_features` is a boolean array-like, "
+                        "the array should be of shape (n_features,). Got "
+                        f"{categorical_features.size} elements while `X` contains "
+                        f"{n_features} features."
+                    )
```

Comment:
Won't this be checked by partial_dependence anyway ? It looks like a double validation

Reply:
Actually, we cannot plot this interaction because we don't have the right display for it but the interaction can be computed. So we need to raise an error only here.

Reply:
As discussed, we have a double validation. The issue here is that the display can accept a list while the `partial_dependence` is only a single feature (or an interaction).

We should revisit the API of `PartialDependenceDisplay` to align it with the API of `partial_dependence` as we do in other displays. It will make the code base much easier to maintain at the cost of calling it several times the display.